lib/purescript-cst/src/Language/PureScript/AST/Exported.hs

```diff
@@ -152,5 +160,14 @@ reorder refs =
   where
   refIndices =
     M.fromList $ zip (map declRefName refs) [(0::Int)..]
-  refIndex decl =
-    declName decl >>= flip M.lookup refIndices
+  refIndex = \case
+    KindDeclaration _ _ n _ ->
+      M.lookup (TyName n) refIndices <|> M.lookup (TyClassName (tyToClassName n)) refIndices
+
+    decl -> declName decl >>= flip M.lookup refIndices
+
+-- |
+-- Workaround to the fact that a `KindDeclaration`'s name's `ProperNameType`
+-- isn't the same as the corresponding `TypeClassDeclaration`'s `ProperNameType`
+tyToClassName :: ProperName 'TypeName -> ProperName 'ClassName
+tyToClassName = coerceProperName
```

Comment:
This function is now accessible outside of `reorder`, which is what I was originally objecting to. This is what I meant:
```suggestion
  -- Workaround to the fact that a `KindDeclaration`'s name's `ProperNameType`
  -- isn't the same as the corresponding `TypeClassDeclaration`'s `ProperNameType`
  tyToClassName :: ProperName 'TypeName -> ProperName 'ClassName
  tyToClassName = coerceProperName
```

Reply:
Yes, I moved it outside of the function because it's now used in two places rather than one. 

Since this is a potentially unsafe function, I'll duplicate the function and move it to a where clause in each place that it's used.

Reply:
Ah right ok. If it's used in two places then I think defining it once at the top level is fine, as long as it's not exported.

Reply:
True, but if it's not defined top-level, then it's less likely to be accidentally exported.

So, I can either define it once and add a note that it shouldn't be exported. Or I can define it locally twice in the places that use it. Which do you prefer?

Reply:
I’d say define it once and don’t bother adding a note. It’s not really a big deal if it does end up being exported; we do coerce type names to class names and back in a bunch of places already, because that’s how type class desugaring works.

Reply:
Ok. I've just removed the last commit that made that change. I believe this PR should be ready to merge if there's no other issues with it.